mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1466,6 +1466,21 @@ func ResourceContainerCluster() *schema.Resource {
 							ValidateFunc:     validation.StringInSlice([]string{"IPV4", "IPV4_IPV6"}, false),
 							Description: 	  `The IP Stack type of the cluster. Choose between IPV4 and IPV4_IPV6. Default type is IPV4 Only if not set`,
 						},
+						"pod_cidr_overprovision_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Description: `Configuration for cluster level pod cidr overprovision. Default is disabled=false.`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"disabled": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+								},
+							},
+						},
```

Comment:
These values will have to be computed 

`				Computed:      true,`..

since they are used by `cluster` resource they need to be read even if they aren't explicitly set

Reply:
I had left out ```Computed``` on purpose because this is an optional and user configurable field.

Do you mean to say that if we refer to these fields in expand and flatten functions, they need to be ```Computed=true```?

Does this also apply for node-pool schema?

Reply:
`google_container_cluster` receives these fields as subfields. They are configured outside of the `google_container_cluster` resource. Thus, all the properties of the `google_container_node_pool` schema have to be `computed`. This will enable them to be read even when not explicitly set. If you notice in the test failures.

```
          # google_container_cluster.cluster must be replaced
        -/+ resource "google_container_cluster" "cluster" {
              ~ cluster_ipv4_cidr           = "10.0.0.0/19" -> (known after apply)
....
        
              ~ ip_allocation_policy {
                  ~ cluster_ipv4_cidr_block       = "10.0.0.0/19" -> (known after apply)
                  ~ services_ipv4_cidr_block      = "10.0.32.0/22" -> (known after apply)
                    # (3 unchanged attributes hidden)
        
                  - pod_cidr_overprovision_config { # forces replacement
                      - disabled = false -> null
                    }
                }
```

`- pod_cidr_overprovision_config { # forces replacement` is being given because terraform is detecting this value isn't set in config. So it wants to unset the value. Thus we are getting a diff that is causing the test failure.

Reply:
Got it. Thanks for the explaination.